airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -551,6 +551,367 @@ def execute(self, context: 'Context') -> str:
+        self.storage_encrypted = storage_encrypted
+        self.kms_key_id = kms_key_id
+        self.domain = domain
+        self.copy_tags_to_snapshot = copy_tags_to_snapshot
+        self.monitoring_interval = monitoring_interval
+        self.monitoring_role_arn = monitoring_role_arn
+        self.domain_iam_role_name = domain_iam_role_name
+        self.promotion_tier = promotion_tier
+        self.timezone = timezone
+        self.enable_iam_database_authentication = enable_iam_database_authentication
+        self.enable_performance_insights = enable_performance_insights
+        self.performance_insights_kms_key_id = performance_insights_kms_key_id
+        self.performance_insights_retention_period = performance_insights_retention_period
+        self.enable_cloudwatch_logs_exports = enable_cloudwatch_logs_exports or []
+        self.processor_features = processor_features or []
+        self.deletion_protection = deletion_protection
+        self.max_allocated_storage = max_allocated_storage
+        self.enable_customer_owned_ip = enable_customer_owned_ip
+        self.custom_iam_instance_profile = custom_iam_instance_profile
+        self.backup_target = backup_target
+        self.network_type = network_type
+
+    def execute(self, context: 'Context') -> str:
+        self.log.info(f"Creating new DB instance {self.db_instance_identifier}")
+        params: Dict[str, Any] = {}
+        if self.db_name:
+            params["DBName"] = self.db_name
+        if self.allocated_storage:
+            params["AllocatedStorage"] = self.allocated_storage
+        if self.master_username:
+            params["MasterUsername"] = self.master_username
+        if self.master_user_password:
+            params["MasterUserPassword"] = self.master_user_password
+        if self.db_security_groups:
+            params["DBSecurityGroups"] = self.db_security_groups
+        if self.vpc_security_group_ids:
+            params["VpcSecurityGroupIds"] = self.vpc_security_group_ids
+        if self.availability_zone:
+            params["AvailabilityZone"] = self.availability_zone
+        if self.db_subnet_group_name:
+            params["DBSubnetGroupName"] = self.db_subnet_group_name
+        if self.preferred_maintenance_window:
+            params["PreferredMaintenanceWindow"] = self.preferred_maintenance_window
+        if self.db_parameter_group_name:
+            params["DBParameterGroupName"] = self.db_parameter_group_name
+        if self.backup_retention_period:
+            params["BackupRetentionPeriod"] = self.backup_retention_period
+        if self.preferred_backup_window:
+            params["PreferredBackupWindow"] = self.preferred_backup_window
+        if self.port:
+            params["Port"] = self.port
+        if self.multi_az:
+            params["MultiAZ"] = self.multi_az
+        if self.engine_version:
+            params["EngineVersion"] = self.engine_version
+        if self.auto_minor_version_upgrade:
```

Comment:
Thank you, nice point!

It looks massive indeed, also we'd need to add a support for new arguments in case of any new introduced in `boto3`. I wasn't sure how to overcome this, but the suggested option looks good. Will fix it 🙂

Reply:
By using the kwargs it solves this issue because we don't need to change anything to support new arguments.